tests/test_lab/test_states/test_ket.py

```diff
@@ -529,6 +529,32 @@ def test_expectation_error(self):
         with pytest.raises(ValueError, match="Expected an operator defined on"):
             ket.expectation(op3)
 
+    def test_expectation_shape_handling(self):
+        cutoff = 200
+        ket = SqueezedVacuum(0, 1.5)
+
+        dm_like = DM.random((0,), max_r=10)
```

Comment:
I guess an equivalent test needs to be added to `test_dm` as well?

Reply:
Also, perhaps `DM.random` shouldn't be used here, to keep the test reproducible. I note that it doesn't always fail here when I remove the changes in `Ket.expectation`.

Lastly, we should likely parametrize the test over the operator. I can push this.